### PR TITLE
Add `.nvmrc` to `doc` directory

### LINKED ISSUE #126: Add `.nvmrc` to `doc` directory
This way we can ensure we always use the same node version locally as in the ci

### PR DESCRIPTION
**Pull request by AnkushinDaniil**
Fixes #126

## Summary by CodeRabbit

- **New Features**
  - Introduced a workflow to automate the build process for the Docusaurus website upon pull requests, enhancing validation of changes.
  - Added a new file to specify the Node.js version for consistent development environments.

- **Chores**
  - Improved the automation flexibility in the build workflow by dynamically reading the Node.js version from the new configuration file.

### CODE REVIEW

**Comment by AnkushinDaniil** [comment]
@joshklop 
review request

**Comment by joshklop** [comment]
@AnkushinDaniil mind rebasing this?

File: doc/.nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+echo "18" > doc/.nvmrc
```
Comment (joshklop) [inline doc/.nvmrc:1]: The version itself should be in the `.nvmrc`

**Review by joshklop — CHANGES_REQUESTED** [review]

File: doc/.nvmrc (hunk 1 above)
Comment (AnkushinDaniil) [reply doc/.nvmrc]: Done

File: .github/workflows/docusaurus.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - uses: actions/checkout@v4
+
+      - name: Read .nvmrc
+        run: echo "NODE_VERSION=$(cat ./doc/.nvmrc)" >> $GITHUB_ENV
+        id: nvm
+
+      - name: Use Node.js ${{ steps.nvm.outputs.NODE_VERSION }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ steps.nvm.outputs.NODE_VERSION }}
```
Comment (joshklop) [inline .github/workflows/docusaurus.yml:22]:
```suggestion
      - uses: actions/setup-node@v4
        with:
          node-version-file: '.nvmrc'
```

**Review by joshklop — CHANGES_REQUESTED** [review]

File: doc/.nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+18
```
[the remark this answers is not in the document]
Comment (AnkushinDaniil) [reply doc/.nvmrc]: Done

**Comment by joshklop** [comment]
We should update the `.github/workflows/docusaurus.yaml` workflow to use `.nvmrc` for the node version.

**Comment by AnkushinDaniil** [comment]
> We should update the `.github/workflows/docusaurus.yaml` workflow to use `.nvmrc` for the node version.

Yes, this happened because of the rebase issue I missed.

File: .github/workflows/docusaurus.yml [hunk 4]
```diff
@@ -11,9 +11,14 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+
+      - name: Get Node.js version from .nvmrc
+        id: node-version
+        run: echo "NODE_VERSION=$(cat ./doc/.nvmrc)" >> $GITHUB_ENV
```
Comment (joshklop) [inline .github/workflows/docusaurus.yml:17]: not necessary

File: .github/workflows/docusaurus.yml [hunk 5]
```diff
@@ -11,9 +11,14 @@ jobs:
...
+
       - uses: actions/setup-node@v4
         with:
-          node-version: 18
+          node-version: ${{ env.NODE_VERSION }}
```
Comment (joshklop) [inline .github/workflows/docusaurus.yml:21]: `node-version-file: .nvmrc`
this should be enough I think

**Review by joshklop — CHANGES_REQUESTED** [review]

File: doc/.nvmrc [hunk 6]
```diff
@@ -0,0 +1 @@
+18.20.4
```
Comment (natebeauregard) [inline doc/.nvmrc:1]: nit: could you add a tailing newline here?

**Pull request merged**